### PR TITLE
kc- added new job to update ALL subjects over a RANGE of quarters

### PR DESCRIPTION
**Pull request by khangvchung**
Added new job that builds on #21 to get all subjects over a range of quarters.

Files modified: JobsController.java, JobsControllerTests.java

Files added: UpdateCourseDataRangeOfQuartersJob.java, UpdateCourseDataRangeOfQuartersJobFactory.java, UpdateCourseDataRangeOfQuartersJobTests.java, UpdateCourseDataRangeOfQuartersJobFactoryTests.java

This can be tested under the jobs-controller updateCoursesRangeOfQuarters

Screen Shot 2022-11-30 at 2 49 02 AM

After executing, you can go under the admin tab and manage jobs to see a screen like this that will continue to list all subjects over a range of quarters.

Screen Shot 2022-11-30 at 2 51 46 AM

Passed all tests locally as well as jacoco and mutation tests.
Screen Shot 2022-11-30 at 5 02 39 PM
Screen Shot 2022-11-30 at 5 02 45 PM

### CODE REVIEW

**Comment by pconrad** [comment]
For the CI/CD failures, see this thread on the team slack.  I think it may help you fix the problem:

https://ucsb-cs156-f22.slack.com/archives/C043AUN4R9B/p1669791453331639

**Review by pikarap1234 — DISMISSED** [review]
LGTM - Shoutout to TAs for merge help

File: src/main/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataRangeOfQuartersJob.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            // end_year = year2;
+
+            // if(qtr1 <= qtr2){
+            //     current_qtr = qtr1;
+            //     end_qtr = qtr2;
+            // } else {
+            //     current_qtr = qtr2;
+            //     end_qtr = qtr1;
+            // }
+        }
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataRangeOfQuartersJob.java:72]: Tell me more about this code.. 

Also why do we still have commented out code in code that's intended to be merged?

File: src/main/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataRangeOfQuartersJob.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                    ctx.log("Courses for [" + subjectArea + " " + quarterYYYYQ + "] have been updated");
+
+                    //break;
+                }
+
+                if(c_year == end_year && c_qtr == end_qtr){
+                    return;
+                }
+
+                //current_qtr++;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataRangeOfQuartersJob.java:122]: Please remove commented out lines

File: src/main/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataRangeOfQuartersJob.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+                if(c_qtr == 4){
+                    //current_year++;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataRangeOfQuartersJob.java:125]: Please remove commented one lines.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please know that it is considered unprofessional to submit a PR with commented out code in it, unless the commented out code is REALLY, REALLY, needed for a very specific reason, and you make a case to the team why it REALLY, REALLY, needs to be kept.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please see my notes on a way to structure this code that is much easier.

**Review by pconrad — APPROVED** [review]
LGTM!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**